observability/log-vs-metric.md

```diff
@@ -0,0 +1,73 @@
+# Logs vs Metrics
+
+## Overview
+
+### Metrics
+
+The purpose of metrics is to inform observers about the health & operations regarding a component or system. A metric represents a point in time measure of a particular source, and data-wise tends to be very small. The compact size allows for efficient collection even at scale in large systems. Metrics also lend themselves very well to preaggregation within the component before collection, reducing computation cost for processing & storing large numbers of metric time series in a central system. Due to how efficiently metrics are processed & stored, it lends itself very well for use in automated alerting, as metrics are an excellent source for the health data for all components in the system.
+
+### Logs
+
+Log data inform observers about the discreet events that occurred within a component or a set of components. Just about every software component log information about its activities over time. This rich data tends to be much larger than metric data and can cause processing issues, especially if components are logging too verbosely. Therefore using log data to understand the health of an extensive system tends to be avoided and depends on metrics for that data. Once metric telemetry highlights potential problem sources, filtered log data for those sources can be used to understand what occurred.
+
+## Formats & Collection Methods
+
+### Logs
+
+When it comes to log collection methods, two of the standard techniques are a direct-write or an agent-based approach.
```

Comment:
The topic of collection is mentioned in the readme.md file, under "Collection": https://github.com/microsoft/code-with-engineering-playbook/blob/e8921dc11baee3e7c32e6cf927f1bcbf2e108356/observability/readme.md#collection

I suggest either consolidating both or just getting rid of the content in the readme file.

Reply:
I'm going to do a root readme.md pass in the next PR. There are a few items that need to be reconciled/updated from there.

Reply:
That's a good idea, @khaines

Reply:
/pillars/logging.md and /pillars/metrics.md also go more in detail on these topics. Some of the collection techniques can also be moved to /pillars/logging.md/metrics as appropriate

Reply:
reorganized some of the content to move pillar specific things into those files directly.